staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode_test.go

```diff
@@ -45,6 +50,194 @@ func TestDecode(t *testing.T) {
 		want          interface{}
 		assertOnError func(t *testing.T, e error)
 	}{
+		{
+			name: "reject duplicate keys (int)",
```

Comment:
Cases for duplicate positive ints and duplicate negative ints would be good, since they are distinct CBOR major types.